apps/emqx/include/logger.hrl

```diff
@@ -40,6 +40,21 @@
     end
 ).
 
+%% NOTE: do not forget to use atom for msg and add every used msg to
+%% the default value of `log.thorttling.msgs` list.
+-define(SLOG_THROTTLE(Level, Data),
+    ?SLOG_THROTTLE(Level, Data, #{})
+).
+
+-define(SLOG_THROTTLE(Level, Data, Meta),
+    case emqx_log_throttler:allow(Level, maps:get(msg, Data)) of
```

Comment:
Doesn't that make
1) `Data` evaluated twice
2) now evaluated even if actual log level is above the passed `Level`?

Reply:
Thanks, fixed: https://github.com/emqx/emqx/pull/12927